### PR TITLE
fix component custom revison loop infinitely create revision

### PR DESCRIPTION
**Pull request by wonderflow**
the status of component update will trigger controller reconcile and create component revision again and again

### CODE REVIEW

File: pkg/controller/core.oam.dev/v1alpha2/applicationconfiguration/revision_enable_test.go [hunk 1]
```diff
@@ -341,3 +343,165 @@ var _ = Describe("Test ApplicationConfiguration Component Revision Enabled trait
...
+
+	BeforeEach(func() {})
+
+	AfterEach(func() {
+		// delete the namespace with all its resources
+		Expect(k8sClient.Delete(ctx, &ns, client.PropagationPolicy(metav1.DeletePropagationForeground))).
+			Should(SatisfyAny(BeNil(), &util.NotFoundMatcher{}))
+	})
+
+	FIt("custom component change revision lead to revision difference, it should not loop infinitely create", func() {
```
Comment (ryanzhang-oss) [inline pkg/controller/core.oam.dev/v1alpha2/applicationconfiguration/revision_enable_test.go:370]: have you tried this test without the fix?

File: pkg/controller/core.oam.dev/v1alpha2/applicationconfiguration/component.go [hunk 2]
```diff
@@ -81,7 +81,7 @@ func (c *ComponentHandler) Generic(_ event.GenericEvent, _ workqueue.RateLimitin
 func isMatch(appConfigs *v1alpha2.ApplicationConfigurationList, compName string) (bool, types.NamespacedName) {
 	for _, app := range appConfigs.Items {
 		for _, comp := range app.Spec.Components {
-			if comp.ComponentName == compName {
+			if comp.ComponentName == compName || (strings.HasPrefix(comp.RevisionName, compName+"-")) {
```
[the remark this answers is not in the document]
Comment (wonderflow) [reply pkg/controller/core.oam.dev/v1alpha2/applicationconfiguration/component.go:84]: we don't have ExtractRevision in this branch

File: pkg/controller/core.oam.dev/v1alpha2/applicationconfiguration/component.go [hunk 3]
```diff
@@ -140,6 +140,10 @@ func newTrue() *bool {
 func (c *ComponentHandler) createControllerRevision(mt metav1.Object, obj runtime.Object) ([]reconcile.Request, bool) {
 	curComp := obj.(*v1alpha2.Component)
 	comp := curComp.DeepCopy()
+	// No generation changed, will not create revision
```
[the remark this answers is not in the document]
Comment (wonderflow) [reply pkg/controller/core.oam.dev/v1alpha2/applicationconfiguration/component.go:143]: sure

File: pkg/controller/core.oam.dev/v1alpha2/applicationconfiguration/revision_enable_test.go (hunk 1 above)
Comment (wonderflow) [reply pkg/controller/core.oam.dev/v1alpha2/applicationconfiguration/revision_enable_test.go:370]: of course

**Pull request merged**